app/models/disk.rb

```diff
@@ -26,6 +26,26 @@ def self.find_cdroms
     where("device_type LIKE '%cdrom%'").to_a
   end
 
+  # A performance improvement was introduced in Rails 6:
+  #
+  #   https://github.com/rails/rails/commit/cc2d614e
+  #
+  # Causes the `present` column in this class to raise the following error:
+  #
+  #   ActiveRecord::DangerousAttributeError:
+  #     present? is defined by Active Record. Check to make sure that you don't
+  #     have an attribute or method with the same name.
+  #
+  # Since there is no whitelist for this method, this attempts to circumvent
+  # that autogenerated error to allow our previously named column to still work
+  # properly.
+  #
+  def self.dangerous_attribute_method?(name)
+    return if name == "present?"
+
+    super
+  end
```

Comment:
Minor, but I'm concerned this could get lost in the file (same with the other ones that are similar)...would it make sense to move it to the top of the file (or at least to be the first method definition - in this case above `find_hard_disks`?   I assume you chose this current location on purpose for a reason.

Moving forward, I assume we want to get rid of this override...how do we go about doing that?

Reply:
> would it make sense to move it to the top of the file...

Yeah, I think doing that is fine.  I can make that change.

> Moving forward, I assume we want to get rid of this override...how do we go about doing that?

The explanation was included in my commit message:

```
[rails6][Disk] Support `present` column

Do to a performance fix in Rails 6:

- rails/rails@cc2d614

The Disk#present column would cause the application to raise the
following error:

  ActiveRecord::DangerousAttributeError:
    present? is defined by Active Record. Check to make sure that you
    don't have an attribute or method with the same name.

This workaround avoids having to create a database migration to fix this
column, and instead allows us to keep the column we have had in place
for years for a performance fix that doesn't apply to this model anyway.
```

But I think I agree that we aren't exactly in a good state here.  The issue at play is that changing the column in the database is infinitely more risky since it might be used in providers and cause hidden bugs in people's automation code (for example).  That is the "Rails way" for fixing it, where we just accept that the change happened, and adjust our columns accordingly.

The other option would be to potentially push back to Rails to add a feature for these kinds of methods where an override can exist for cases like this, where the said class method is just a performance fix, and the true intent of something like that warning is to avoid naming columns with methods like `save` and `update`.  That said, not sure of what that would look like and if the Rails core team would be receptive to something like that, but also something we can't bank on waiting for a change here.